examples/k8s/hz-go-service/service.go

```diff
@@ -0,0 +1,347 @@
+	sampleMap = "myDistributedMap"
+)
+
+var (
+	config    *ServiceConfig
+	imdg      *InMemoryHzStorage
+	hzc       *hazelcast.Client
+	endpoints = [...]string{"config", "map"}
+)
+
+// Info return body json structure of request on map handler
+type Info struct {
+	Msg string `json:"msg"`
+}
+
+// InMemoryHzStorage Hazelcast storage used by the service.
+type InMemoryHzStorage struct {
+	myMap *hazelcast.Map
+}
+
+// ServiceConfig includes information about service configuration.
+type ServiceConfig struct {
+	ServiceName string        `json:"serviceName"`
+	Port        int           `json:"port"`
+	Timeout     time.Duration `json:"timeout"`
+}
+
+// getConfigHandler handles incoming config endpoint, returns status of service configs.
+func getConfigHandler(rw http.ResponseWriter, r *http.Request) {
+	rw.Header().Add("Content-Type", "application/json")
+	rw.WriteHeader(http.StatusOK)
+	serviceConfigResponse := struct {
+		ServiceConfig
+		hz.ClientInfo
+	}{*config, hz.ClientInfo{ClientName: hzc.Name(), ClientRunning: hzc.Running(), MapSize: 0}}
+	var err error
+	serviceConfigResponse.ClientInfo.MapSize, err = imdg.myMap.Size(r.Context())
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("Hazelcast map size, err: %v", err), http.StatusInternalServerError)
+		return
+	}
+	rspJson, err := json.Marshal(serviceConfigResponse)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("Json parsing, err: %v", err), http.StatusBadRequest)
+		return
+	}
+	_, err = rw.Write(rspJson)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("Could not write json data, err: %v", err), http.StatusInternalServerError)
+		return
+	}
+}
+
+// validateURLQueryParameter validates whether given request url query parameter is valid.
+func validateURLQueryParameter(r *http.Request) (mapNames, keys []string, err error) {
+	// resolve query parameters
+	if err := r.ParseForm(); err != nil {
+		return nil, nil, err
+	}
+	var ok bool
+	if mapNames, ok = r.Form["name"]; !ok {
+		return nil, nil, errors.New("there is no query parameter as name")
+	}
+	if keys, ok = r.Form["key"]; !ok {
+		return nil, nil, errors.New("there is no query parameter as key")
+	}
+	return mapNames, keys, nil
+}
+
+// mapEntryHandler make put and get operation on map.
+func mapEntryHandler(rw http.ResponseWriter, r *http.Request) {
+	// set response type
+	rw.Header().Set("Content-Type", "application/json")
+	var err error
+	mapNames, keys, err := validateURLQueryParameter(r)
+	if err != nil {
+		http.Error(rw, fmt.Sprintf("URL query parameters cannot be validated, err: %v", err), http.StatusBadRequest)
+		return
+	}
+	mapName := mapNames[0]
+	key := keys[0]
+	// wrap a timeout context for the hazelcast operation
+	ctx, cancel := context.WithTimeout(r.Context(), hzTimeout)
+	defer cancel()
+	objInfo, err := hzc.GetDistributedObjectsInfo(ctx)
+	if err != nil {
+		http.Error(rw, "Hazelcast objects info cannot be obtained", http.StatusInternalServerError)
+		return
+	}
+	exist := false
+	for _, o := range objInfo {
```

Comment:
How would you feel about simplifying this? For example we can create the map even if it does not exist. That would eliminate quite a bit of branching.

Reply:
I couldn't get what you mean 😞